python/example_code/lookoutvision/inference.py

```diff
@@ -0,0 +1,164 @@
+import os
+import boto3
+
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger(__name__)
+
+
+class Inference:
+    """
+    Shows how to detect anomalies in an image using a trained Amazon Lookout
+    for Vision model.
+    """
+    @staticmethod
+    def detect_anomalies(lookoutvision_client, project_name, model_version, photo):
+        """
+        Detects anomalies in an image (jpg/png) by using your Amazon Lookout for Vision model.
+        param: lookoutvision_client: An Amazon Lookout for Vision boto3 client.
+        param: project_name: The name of the project that contains the model that you want to use.
+        param: model_version: The version of the model that you want to use.
+        param: photo: The path and name of the image in which you want to detect anomalies.
+        """
+
+        try:
+
+            image_type = imghdr.what(photo)
+            content_type = ""
+
+            if image_type == "jpeg":
+                content_type = "image/jpeg"
+            elif image_type == "png":
+                content_type = "image/png"
+            else:
+                logger.info("Invalid image type for %s", photo)
+                raise ValueError(
+                    "Invalid file format. Supply a jpeg or png format file: {}".format(photo)
+                )
+
+            # Call detect_anomalies
+            logger.info("Detecting anomalies in %s", photo)
+            with open(photo, "rb") as image:
+                response = lookoutvision_client.detect_anomalies(
+                    ProjectName=project_name,
+                    ContentType=content_type,  # "image/jpeg" or image/png
+                    Body=image.read(),
+                    ModelVersion=model_version,
+                )
+            anomalous = response["DetectAnomalyResult"]["IsAnomalous"]
+            confidence = response["DetectAnomalyResult"]["Confidence"]
+
+            logger.info("Anomalous?: %s", format(anomalous))
+            logger.info("Confidence: %s", format(confidence))
+            return anomalous, confidence
+
+        except FileNotFoundError as err:
+            logger.info("Couldn't find file: %s", photo)
+            raise
+
+        except ClientError as err:
+            logger.info(format(err))
+            raise
+
+    @staticmethod
+    def download_from_s3(photo):
+        """
+        Downloads an image from an S3 bucket.
+        param: photo: The S3 path of a photo to download.
+        return: The local path to the downloaded file.
+        """
+
+        try:
+
+            bucket, key = photo.replace("s3://", "").split("/", 1)
+            local_file = os.path.basename(photo)
+        except ValueError as err:
+            logger.info("Couldn't get S3 info for %s: %s", photo, format(err))
+            raise ValueError("Couldn't get S3 info for {}.".format(photo)) from err
+
+        try:
+            logger.info("Downloading %s", photo)
+            s3_client = boto3.client("s3")
```

Comment:
Pass s3_resource as an argument instead of creating locally.

Reply:
Fixed.